pkg/logs/input/eventlog/launcher.go

```diff
@@ -0,0 +1,92 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package eventlog
+
+import (
+	log "github.com/cihub/seelog"
+
+	"github.com/DataDog/datadog-agent/pkg/logs/auditor"
+	"github.com/DataDog/datadog-agent/pkg/logs/config"
+	"github.com/DataDog/datadog-agent/pkg/logs/pipeline"
+	"github.com/DataDog/datadog-agent/pkg/logs/restart"
+)
+
+// Launcher is in charge of starting and stopping windows event logs tailers
+type Launcher struct {
+	sources          []*config.LogSource
+	pipelineProvider pipeline.Provider
+	auditor          *auditor.Auditor
+	tailers          map[string]*Tailer
+}
+
+// New returns a new Launcher.
+func New(sources []*config.LogSource, pipelineProvider pipeline.Provider, auditor *auditor.Auditor) *Launcher {
+	windowsEventSources := []*config.LogSource{}
+	for _, source := range sources {
+		if source.Config.Type == config.WindowsEventLogType {
+			windowsEventSources = append(windowsEventSources, source)
+		}
+	}
+	return &Launcher{
+		sources:          windowsEventSources,
+		pipelineProvider: pipelineProvider,
+		auditor:          auditor,
+		tailers:          make(map[string]*Tailer),
+	}
+}
+
+// Start starts new tailers.
+func (l *Launcher) Start() {
+	log.Info("Start tailing eventlog")
+	availableChannels, err := EnumerateChannels()
+	if err != nil {
+		log.Debug("Could not list eventlog channels: ", err)
```

Comment:
Today we don't need this logic to enumerate channels. We need to specify each channel in the config.
I let this code in debug mode so one can list all the available channels, I thought that we didn't need to report the error neither.

This might change in the future depending on user feedback